pinot-common/src/thrift/request.thrift

```diff
@@ -152,6 +153,7 @@ struct BrokerRequest {
  14: optional map<string, string> queryOptions;
  15: optional HavingFilterQuery havingFilterQuery;
  16: optional HavingFilterQueryMap havingFilterSubQueryMap;
+ 17: optional query.PinotQuery pinotQuery;
```

Comment:
Why do we put pinotQuery into BrokerRequest? All the info should be in BrokerRequest right? This will double the size of the serialized bytes

Reply:
No, pinotQuery is the new structure that will allow us to support full sql functionality.In subsequent PR's, Query execution layer will be changed to start using PinotQuery instead of existing fields in broker request such as selection, aggregation etc.

Regarding the size, yes it will be double in the validation/migration phase. But once we start using pinotquery, we will not use other fields in brokerRequest.

Reply:
The size of data transferred from broker to server will also double. I wound recommend using a separate endpoint for PinotQuery instead of reuse the endpoint for BrokerRequest. The query execution layer should take either BrokerRequest or PinotQuery, but not PinotQuery nested inside BrokerRequest.

Reply:
separate endpoint is fine. Its more about how to change the execution layer. BrokerRequest is used everywhere in the execution layer. Changing that to pinotquery will be very invasive. Let me clone the issue from the fork and we can discuss the high-level steps.

Reply:
Sure, we can discuss under the issue